search-service/src/main/kotlin/com/egm/stellio/search/util/TemporalQueryUtils.kt

```diff
@@ -36,17 +37,20 @@ fun parseAndCheckQueryParams(
         Optional.ofNullable(requestParams.getFirst(QUERY_PARAM_OPTIONS)),
         OptionsParamValue.AUDIT
     )
-    val temporalQuery = buildTemporalQuery(requestParams, inQueryEntities)
+    val temporalQuery = buildTemporalQuery(requestParams, inQueryEntities).bind()
 
-    return TemporalEntitiesQuery(
+    TemporalEntitiesQuery(
         queryParams = queryParams,
         temporalQuery = temporalQuery,
         withTemporalValues = withTemporalValues,
         withAudit = withAudit
     )
```

Comment:
there shouldn't be `.right` at the end ?

Reply:
no it is in a either effect (and actually compilation would have failed if the return type was not correct)